scripts/k3s-versions.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# This script expects to be called from the root of the repo.
+# It will rebuild resources/k3s-versions.json from both the k3s update
+# channel and the GitHub k3s releases list.
+# Creates a pull request if the new version is different.
+
+set -eu
+
+K3S_VERSIONS="resources/k3s-versions.json"
+BRANCH_NAME="gha-update-k3s-versions"
+NEW_PR="true"
+
+if git rev-parse --verify "origin/${BRANCH_NAME}" 2>/dev/null; then
```

Comment:
Given that we have similar code elsewhere, it might be better to make this TypeScript and make the scripts share code instead.
https://github.com/rancher-sandbox/rancher-desktop/blob/f692d778245b4bffd637f378403c038debe615e3/scripts/release-merge-to-main.ts#L99-L106

Reply:
Not really keen to rewrite a single line of shell code into 20 lines of TypeScript that becomes impossible to debug. FWIW, I think the existing code for e.g. `rddepman` is considerable less readable than the simple script.

Reply:
That simple script has an unreadable JQ expression that didn't actually work in CI… https://github.com/mook-as/rd/actions/runs/12286322662/job/34286221124

(I was testing dropping the `depth:` bit, but it didn't get that far. I assume the version of `jq` the runner has installed is too old/new.)